code/game/objects/objs.dm

```diff
@@ -109,15 +109,18 @@
 	if((allow_pass_flags & PASS_DEFENSIVE_STRUCTURE) && (mover.pass_flags & PASS_DEFENSIVE_STRUCTURE))
 		return TRUE
 
-	if((allow_pass_flags & PASS_LOW_STRUCTURE) && (mover.pass_flags & PASS_LOW_STRUCTURE))
-		return TRUE
-
 	if((allow_pass_flags & PASS_GLASS) && (mover.pass_flags & PASS_GLASS))
 		return TRUE
 
 	if(mover?.throwing && (allow_pass_flags & PASS_THROW))
 		return TRUE
 
+	if((allow_pass_flags & PASS_LOW_STRUCTURE) && (mover.pass_flags & PASS_LOW_STRUCTURE))
+		return TRUE
+
+	if((allow_pass_flags & PASS_WALKOVER) && ismob(mover) && SEND_SIGNAL(target, COMSIG_OBJ_TRY_ALLOW_THROUGH))
```

Comment:
This is generally bad design, signals are not replacements for procs, they are to be used inside of existing generic behaviors.

Reply:
I'm all ears if you have another suggestion, I'm just trying to clear out snowflake code, and loops through turf contents.

Reply:
atleast follow the convention of positive return values (COMPONENT_DENY_ENTRY) interrupting action instead of the other way around